src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+    }
+  }
+
+  private static final Instant TODAY = ZonedDateTime.now().toInstant();
+  private static final Instant TWO_DAYS_AGO = ZonedDateTime.now().minusDays(2).toInstant();
+
+  private String project;
+  private String start;
+  private String end;
+  private String analyte;
+  private final String todate = removeTime(TODAY.toString());
+
+  private static final Option OPT_PROJECT = CommonOptions.project(true);
+  private static final Option OPT_AFTER = CommonOptions.after(true);
+  private static final Option OPT_BEFORE = CommonOptions.before(true);
+  private static final Option OPT_ANALYTE = CommonOptions.analyte(false);
+  public static final String REPORT_NAME = "project-status";
+
+  // keep track of the base stock name for all samples/libraries/runs active this month
+  Map<String, ReportItem> dnaThisMonth = Maps.newTreeMap();
+  Map<String, ReportItem> rnaThisMonth = Maps.newTreeMap();
+  Map<String, ReportItem> allDna = Maps.newTreeMap();
+  Map<String, ReportItem> allRna = Maps.newTreeMap();
+  List<Map.Entry<String, ReportItem>> dnaRecentList;
+  List<Map.Entry<String, ReportItem>> rnaRecentList;
+  List<Map.Entry<String, ReportItem>> dnaAllList;
+  List<Map.Entry<String, ReportItem>> rnaAllList;
+
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    return Sets.newHashSet(OPT_PROJECT, OPT_AFTER, OPT_BEFORE, OPT_ANALYTE);
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    if (cmd.hasOption(OPT_AFTER.getLongOpt())) {
+      String after = cmd.getOptionValue(OPT_AFTER.getLongOpt());
+      if (!after.matches(DATE_REGEX)) {
+        throw new ParseException("After date must be in format yyyy-mm-dd");
+      }
+      this.start = after;
+    }
+
+    if (cmd.hasOption(OPT_BEFORE.getLongOpt())) {
+      String before = cmd.getOptionValue(OPT_BEFORE.getLongOpt());
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+
+    if (cmd.hasOption(OPT_ANALYTE.getLongOpt())) {
+      String analyte = cmd.getOptionValue(OPT_ANALYTE.getLongOpt());
+      if (!DNA.equals(analyte) && !RNA.equals(analyte)) {
+        throw new ParseException("Analyte must be DNA or RNA");
+      }
+      this.analyte = analyte;
+    }
+
+    this.project = cmd.getOptionValue(OPT_PROJECT.getLongOpt());
+  }
+
+  @Override
+  public String getTitle() {
+    return "Project Status Report (Gazpacho) for project " + project
+        + (analyte == null ? "" : (DNA.equals(analyte) ? " DNA" : " RNA"))
+        + " for "
+        + (start == null ? "Any Time" : start)
+        + " - "
+        + (end == null ? "Now" : end)
+        + ". Generated " + new SimpleDateFormat(DATE_FORMAT).format(new Date());
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException, IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 69 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)